### PR TITLE
API changes now check that API pages render

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/773. 

Now that we've fixed all historical pages not rendering, it's important we don't regress. The check is much faster than it used to be, so we decided it's worth the CI slowdown.

More broadly, it's become clear that there are two types of changes we make: a) API docs, and b) non-API docs. They are very different workflows, since API docs are 100% automated. Our CI should better reflect this reality, whereas before for example `main.yml` would check relative links for API docs. 

To speed up the API CI job, we split it up into two parallel jobs.

Once we start working more with translations, we'll want to consider a `translations.yml` workflow. Those are also fairly isolated in comparison to API docs and non-API docs.

### CODE REVIEW

File: .github/workflows/api-checks.yml [hunk 1]
```diff
@@ -10,14 +10,11 @@
...
-# This Action runs checks things that we care about but would be
-# too slow to check in every PR.
-
-name: Extended checks
+name: API checks
 on:
   workflow_dispatch:
   schedule:
-    - cron: "0 0 * * *" # Every day at midnight UTC
+    - cron: "0 0 * * 0,2,4" # Every Sunday, Tuesday, and Thursday at midnight UTC
```
Comment (Eric-Arellano) [inline .github/workflows/api-checks.yml:17]: Switched to 3 times a week to waste fewer resources. Most people aren't checking on Friday night and Saturday what happened, for example.

**Review by Eric-Arellano — COMMENTED** [review]
See https://github.com/Qiskit/documentation/actions/runs/7850053208/job/21424707125 and https://github.com/Qiskit/documentation/actions/runs/7850053208/job/21424707277 for examples.

**Review by frankharkins — APPROVED** [review]
Looks good, thanks!

**Review by arnaucasau — APPROVED** [review]
LGTM, thank you Eric!

**Pull request merged**